spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  describe 'POST #create' do
+    context 'with valid parameters' do
+      it 'creates a new post' do
+        expect {
+          post :create, params: { post: { title: 'Test Title', description: 'Test Description' } }
+        }.to change(Post, :count).by(1)
+      end
+
+      it 'returns a successful response' do
+        post :create, params: { post: { title: 'Test Title', description: 'Test Description' } }
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context 'with invalid parameters' do
+      it 'does not create a new post' do
+        expect {
+          post :create, params: { post: { title: nil, description: 'Test Description' } }
+        }.not_to change(Post, :count)
+      end
+
+      it 'returns unprocessable_entity status' do
+        post :create, params: { post: { title: nil, description: 'Test Description' } }
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+    end
+  end
+
+  describe 'GET #index' do
+    it 'returns a list of posts' do
+      # Create posts directly in the database
+      Post.create(title: 'Title 1', description: 'Description 1')
```

Comment:
Use Factory to create data

Reply:
I have inlcuded this in `feature/authentication-using-jwt-token` branch